examples/data-objects/todo/src/TodoItem/TodoItemView.tsx

```diff
@@ -22,33 +22,40 @@ interface TodoItemViewProps {
 
 export const TodoItemView: React.FC<TodoItemViewProps> = (props: TodoItemViewProps) => {
 	const { todoItemModel, className } = props;
-	const itemText = todoItemModel.getText();
-	const [checked, setChecked] = useState<boolean>(todoItemModel.getCheckedState());
+
+	const [itemTitle, setItemTitle] = useState<SharedString | undefined>(undefined);
+	const [itemDescription, setItemDescription] = useState<SharedString | undefined>(undefined);
 	const [detailsVisible, setDetailsVisible] = useState<boolean>(false);
 
+	useTree(todoItemModel);
+
 	useEffect(() => {
-		const refreshCheckedStateFromModel = () => {
-			setChecked(todoItemModel.getCheckedState());
-		};
-		todoItemModel.on("checkedStateChanged", refreshCheckedStateFromModel);
-		refreshCheckedStateFromModel();
+		void Promise.resolve(todoItemModel.title.get()).then((text) => {
+			setItemTitle(text as SharedString);
+		});
+	}, [todoItemModel]);
 
-		return () => {
-			todoItemModel.off("checkedStateChanged", refreshCheckedStateFromModel);
-		};
+	useEffect(() => {
+		void Promise.resolve(todoItemModel.description.get()).then((text) => {
+			setItemDescription(text as SharedString);
+		});
 	}, [todoItemModel]);
```

Comment:
Nit: since this effect only uses `todoItemMode.description`
```suggestion
	}, [todoItemModel.description]);
```

Reply:
updated with suggested change